kikuchipy/util/experimental.py

```diff
@@ -248,9 +250,111 @@ def _adaptive_histogram_equalization_chunk(
         equalized_patterns[nav_idx] = _rescale_pattern(
             equalized_pattern, dtype_out=dtype_in
         )
+
     return equalized_patterns
 
 
+def _average_neighbour_patterns_chunk(
+    patterns, n_averaged, kernel, dtype_out=None
+):
+    """Average neighbour pattern intensities inplace within a
+    square kernel, within a chunk.
+
+    Parameters
+    ----------
+    patterns : dask.array.Array
+        Patterns to average, with some overlap with surrounding chunks.
+    n_averaged : dask.array.Array
+        Number of neighbour patterns that will be averaged with
+    kernel : numpy.ndarray
+        Averaging kernel.
+    dtype_out : numpy.dtype, optional
+        Data type of corrected patterns. If ``None`` (default), it is
+        set to the same data type as the input patterns.
+
+    Returns
+    -------
+    averaged_patterns : dask.array.Array
+        Averaged patterns.
+    """
+
+    if dtype_out is None:
+        dtype_out = patterns.dtype
+
+    # Convolve patterns with averaging kernel
+    convolved_patterns = convolve(
+        patterns.astype(np.float32), weights=kernel, mode="constant", cval=0,
+    )
+
+    # Divide convolved patterns by number of neighbours averaged with
+    averaged_patterns = np.empty_like(convolved_patterns, dtype=dtype_out)
+    for nav_idx in np.ndindex(patterns.shape[:-2]):
+        averaged_patterns[nav_idx] = (
+            convolved_patterns[nav_idx] / n_averaged[nav_idx]
+        ).astype(dtype_out)
+
+    return averaged_patterns
+
+
+def _pattern_kernel(axes, n_neighbours=1, exclude_kernel_corners=True):
+    """Create a pattern kernel with a given number of neighbours,
+    possibly excluding corner patterns.
+
+    Parameters
+    ----------
+    axes : hyperspy.axes.AxesManager
+        A HyperSpy signal axes manager containing navigation and signal
+        dimensions and shapes.
+    n_neighbours : int, optional
+        Number of neighbours on each side to average with, default
+        is 1.
```

Comment:
"number of neighbours" might be misunderstood and I think a more specific description could be useful. New users might be confused if n_neighbours is the total number of neighbouring nav pixels used in the averaging, if n_neighbours denotes the distance from the centre pixel to the other pixels that are used, or if n_neighbours is the order of nearest neighbours that are included. In the documentation, more info is given (raidal distance), and it could be useful to include that here and for average_neighbour_patterns().

Reply:
The docstring has now been updated, could you perhaps have a quick look?